packages/schemas/src/multiVariableText/propPanel.ts

```diff
@@ -0,0 +1,139 @@
+
+  // The wrapping form element has a display:flex which limits the width of the form fields, removing.
+  (rootElement.parentElement as HTMLElement).style.display = 'block';
+
+  if (varNames.length > 0) {
+    for (let variableName of varNames) {
+      const varRow = placeholderRowEl.cloneNode(true) as HTMLElement;
+
+      const textarea = varRow.querySelector('textarea') as HTMLTextAreaElement;
+      textarea.id = 'dynamic-var-' + variableName;
+      textarea.value = variables[variableName];
+      textarea.addEventListener('change', (e: Event) => {
+        variables[variableName] = (e.target as HTMLTextAreaElement).value;
+        changeSchemas([{ key: 'content', value: JSON.stringify(variables), schemaId: activeSchema.id }]);
+      });
+
+      const label = varRow.querySelector('label') as HTMLLabelElement
+      label.innerText = variableName;
+
+      varRow.style.display = 'block';
+      rootElement.appendChild(varRow);
+    }
+  } else {
+    const para = document.createElement('p');
+    para.innerHTML = i18n('schemas.mvt.typingInstructions')
+        + ` <code style="color:${options?.theme?.token?.colorPrimary || "#168fe3"}; font-weight:bold;">{`
+        + i18n('schemas.mvt.sampleField')
+        + '}</code>';
+    rootElement.appendChild(para);
+  }
+};
+
+export const propPanel: PropPanel<MultiVariableTextSchema> = {
+  schema: (propPanelProps: Omit<PropPanelWidgetProps, 'rootElement'>) => {
+    if (typeof parentPropPanel.schema !== 'function') {
+      throw Error('Oops, is text schema no longer a function?');
+    }
+    return {
+      ...parentPropPanel.schema(propPanelProps),
+      '-------': { type: 'void', widget: 'Divider' },
+      dynamicVarContainer: {
+        title: propPanelProps.i18n('schemas.mvt.variablesSampleData'),
+        type: 'string',
+        widget: 'Card',
+        span: 24,
+        properties: {
+          dynamicVariables: {
+            type: 'object',
+            widget: 'mapDynamicVariables',
+            bind: false,
+            span: 24
+          },
+          placeholderDynamicVar: {
+            title: 'Placeholder Dynamic Variable',
+            type: 'string',
+            format: 'textarea',
+            props: {
+              id: 'placeholder-dynamic-var',
+              autoSize: {
+                minRows: 2,
+                maxRows: 5,
+              },
+            },
+            span: 24,
+          },
+        }
+      },
+
+    };
+  },
+  widgets: { ...parentPropPanel.widgets, mapDynamicVariables },
+  defaultSchema: {
+    ...parentPropPanel.defaultSchema,
+    type: 'multiVariableText',
+    icon: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-type"><polyline points="4 7 4 4 20 4 20 7"/><line x1="9" x2="15" y1="20" y2="20"/><line x1="12" x2="12" y1="4" y2="20"/></svg>',
```

Comment:
I used the standard `T` text icon.
I chose this because if you are using this schema then you don't need Read-Only Text, since it can support this functionality. Therefore, these 2 wouldn't be used together and the `T` is clear.

![Screenshot 2024-07-18 at 15 31 00](https://github.com/user-attachments/assets/331c9601-537c-46bf-8e29-e3fcc1c3cb80)

Reply:
May I ask what is this `icon` property in schema definitions used for? Could not find it in the docs.